server/src/internalClusterTest/java/org/elasticsearch/threadpool/SimpleThreadPoolIT.java

```diff
@@ -95,4 +111,66 @@ public void testThreadNames() throws Exception {
         }
     }
 
+    public void testThreadPoolMetrics() throws Exception {
+        internalCluster().startNode();
+
+        final String dataNodeName = internalCluster().getRandomNodeName();
+        final TestTelemetryPlugin plugin = internalCluster().getInstance(PluginsService.class, dataNodeName)
+            .filterPlugins(TestTelemetryPlugin.class)
+            .findFirst()
+            .orElseThrow();
+
+        logger.info("do some indexing, flushing, optimize, and searches");
+        int numDocs = randomIntBetween(2, 100);
+        IndexRequestBuilder[] builders = new IndexRequestBuilder[numDocs];
+        for (int i = 0; i < numDocs; ++i) {
+            builders[i] = prepareIndex("idx").setSource(
+                jsonBuilder().startObject()
+                    .field("str_value", "s" + i)
+                    .array("str_values", new String[] { "s" + (i * 2), "s" + (i * 2 + 1) })
+                    .field("l_value", i)
+                    .array("l_values", new int[] { i * 2, i * 2 + 1 })
+                    .field("d_value", i)
+                    .array("d_values", new double[] { i * 2, i * 2 + 1 })
+                    .endObject()
+            );
+        }
+        indexRandom(true, builders);
+        int numSearches = randomIntBetween(2, 100);
+        for (int i = 0; i < numSearches; i++) {
+            assertNoFailures(prepareSearch("idx").setQuery(QueryBuilders.termQuery("str_value", "s" + i)));
+            assertNoFailures(prepareSearch("idx").setQuery(QueryBuilders.termQuery("l_value", i)));
+        }
+        plugin.collect();
```

Comment:
@stu-elastic this is how we should trigger the collection right?
I was surprised we did not have this method before, perhaps other metrics were not testing this?